pkg/crosscluster/logical/replication_statements.go

```diff
@@ -119,16 +132,31 @@ func newInsertStatement(
 // newMatchesLastRow creates a WHERE clause for matching all columns of a row.
 // It returns a tree.Expr that compares each column to a placeholder parameter.
 // Parameters are ordered by column ID, starting from startParamIdx.
-func newMatchesLastRow(columns []catalog.Column, startParamIdx int) (tree.Expr, error) {
+func newMatchesLastRow(columns []columnSchema, startParamIdx int) (tree.Expr, error) {
 	var whereClause tree.Expr
 	for i, col := range columns {
-		placeholder, err := newTypedPlaceholder(startParamIdx+i, col)
+		if col.isComputed {
+			// Skip computed columns since they are not needed to fully specify the
+			// contents of the row.
```

Comment:
If this is also skipping over placeholder IDs, then I don't think it will work for the same reason as mentioned above: the type checker won't be able to determine the type for the skipped placeholders.

In this case, I think it's probably harmless to add the computed columns to the WHERE clause. We should still get the fast paths.